### ISSUE TITLE
Support Tone-based Surfaces in Material 3

### ISSUE THREAD

**Issue by jamiewest** [flutter/flutter#125807]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

The material design guidelines have added additional surface color roles that are tone based. These are currently not available in the [ColorScheme](https://github.com/flutter/flutter/blob/d9ea36ccb7a81cf0bcff375e92c6d8fffa5f98cf/packages/flutter/lib/src/material/color_scheme.dart) class.

<img width="1024" alt="unnamed" src="https://user-images.githubusercontent.com/712135/235486880-ebacc851-2cdf-4ec4-a05d-5b68c8065fef.png">

Blog - [https://material.io/blog/tone-based-surface-color-m3](https://material.io/blog/tone-based-surface-color-m3)
Docs - [https://m3.material.io/styles/color/the-color-system/color-roles#0abbf8b7-61e1-49ee-9f97-4967beb1e4fe](https://m3.material.io/styles/color/the-color-system/color-roles#0abbf8b7-61e1-49ee-9f97-4967beb1e4fe)
Tokens - [https://m3.material.io/styles/color/the-color-system/tokens#7fd4440e-986d-443f-8b3a-4933bff16646](https://m3.material.io/styles/color/the-color-system/tokens#7fd4440e-986d-443f-8b3a-4933bff16646)

### Proposal

Add additional surface color roles to the `ColorScheme` class, and generate the colors when provided a seed value, otherwise default values will apply as they are stated in the tokens section above.

**Comment by jamiewest** [comment]
I have reviewed this a bit more and I don't like that there is a generated `Scheme` class used from the material_color_utilities package is paired to the `ColorScheme` here in the framework. We could generate the light and dark `CorePalete`'s here and set the individual color's from the palete. 

This is mock of what could be changed here `ColorScheme.fromSeed` 
```dart
    final CorePalette palette;
    palette = CorePalette.of(seedColor.value);

    switch (brightness) {
      case Brightness.light:
        return ColorScheme(
          primary: primary ?? Color(palette.primary.get(40)),
          onPrimary: onPrimary ?? Color(palette.primary.get(100)),
          primaryContainer: primaryContainer ?? Color(palette.primary.get(90)),
```

This would avoid having to wait on the material_color_utilities package to update their `Scheme` class and push an update.

**Comment by TahaTesser** [comment]
duplicate of https://github.com/flutter/flutter/issues/115912
